docs/developers/quickstart/verify-smart-contract/truffle.md

```diff
@@ -0,0 +1,98 @@
+---
+title: Truffle
+sidebar_position: 1
+---
+
+To verify your contracts, you can use Truffle's nifty plugin [truffle-plugin-verify](https://github.com/rkalis/truffle-plugin-verify)!
+
+These steps assume you've been storing your secret keys in a `.env` file, which you can read more about [here](../deploy-smart-contract/truffle.md/#truffle-configjs).
+
+## Download the plugin
+
+In your Truffle project, install the plugin:
+
+```bash
+npm install -D truffle-plugin-verify
+```
+
+And add the plugin to `truffle-config.js`:
+
+```javascript
+module.exports = {
+  /* ... rest of truffle-config */
+
+  plugins: ["truffle-plugin-verify"],
+};
+```
+
+## Add your Etherscan API Key
+
+Then, you'll need to get an Etherscan key by creating an account at https://etherscan.io/. Grab your key, and add it to the `.env`:
+
+```
+MNEMONIC=YOUR_MNEMONIC_HERE
+INFURA_API_KEY=YOUR_API_KEY_HERE
+ETHERSCAN_API_KEY=YOUR_API_KEY_HERE
+```
+
+## Add the custom chain
+
+Because ConsenSys zkEVM is not supported by the network yet, we'll have to get our Etherscan API key and add a custom chain like so:
+
+```javascript
+require("dotenv").config();
+const { MNEMONIC, INFURA_API_KEY, ETHERSCAN_API_KEY } = process.env;
+// ... rest of truffle-config
+module.exports = {
+  networks: {
+    consensyszkevmgoerli: {
+      provider: () => {
+        return new HDWalletProvider(
+          MNEMONIC,
+          `https://consensys-zkevm-goerli-prealpha.infura.io/v3/${INFURA_API_KEY}`,
+        );
+      },
+      verify: {
+        apiUrl: "https://explorer.goerli.zkevm.consensys.net/api",
```

Comment:
Yeah I tried it without

The verifiers look for the etherscan api key, so it fails fast looking for it

Reply:
Ok, that's probably a bug on their side then 🤷🏼 . Maybe putting a placeholder then to avoid having users go on Etherscan to find an API key that won't be used in the long term?

Reply:
I haven't had time to test with a fake etherscan key so I'm gonna create an issue to try it later